pkg/v1/validate/image.go

```diff
@@ -113,12 +114,25 @@ func validateLayers(img v1.Image, opt ...Option) error {
 		return layersExist(layers)
 	}
 
+	cf, err := img.ConfigFile()
```

Comment:
I believe these calls were intentionally after processing layers to make this work with streaming layers. We could call `img.Manifest()` early only if we hit `io.ErrUnexpectedEOF` to generate a better error?

Reply:
I don't think that calling `img.ConfigFile()` earlier will result in different behavior or errors, because the result of `img.Layers()`, which was previously called, should always be based on a successfully read/parsed config file for the image.

However, I'm not sure if there's a similar guarantee for the result of `img.Manifest()`, so your suggestion makes sense.